swap_meet/vendor.py

```diff
@@ -1,2 +1,120 @@
 class Vendor:
-    pass
\ No newline at end of file
+# WAVE 1
+    def __init__(self, inventory=None):
+        # if inventory is none then set an empty list by default
+        if inventory is None:
+            inventory = []
+        # attribute (allows us to access inventory later)
+        self.inventory = inventory
+
+
+    # add method
+    def add(self, item):
+        # append the item to the inventory and return item
+        self.inventory.append(item)
+        return item
+    # remove method
+    def remove(self, item):
+        # check if item is not in inventory rhen return False
+        if item not in self.inventory:
+            return False
+        else:
+            # remove the item from the inventory using pop method
+            # (first get index of item) then return the removed item
+            index = self.inventory.index(item)
+            removed_item = self.inventory.pop(index)
+            return removed_item
```

Comment:
👍 Nice approach to look for the case that would prevent us from removing an item and handling that before getting to our main logic. This guards the main logic from a data configuration that would cause a problem.

Note that since this branch `return`s from the function, any following code is effectively the "false" branch. This means the code currently in the false branch could be unindented from beneath the else, and the else removed.

```py
        if item not in self.inventory:
            return False

        # "false" code unindented
        index = self.inventory.index(item)
        removed_item = self.inventory.pop(index)
        return removed_item
```

This approach is very common, so we should get used to expressing our logic this way.